drivers/include/pulse_counter.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2017 UC Berkeley
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_pulse_counter sensor
+ * @ingroup     drivers_sensors
+ *
+ * The connection between the MCU and the PULSE_COUNTER is based on the
+ * GPIO-interface.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Driver for the PULSE_COUNTER sensor
+ *
+ * @author      Hyung-Sin <hs.kim@cs.berkeley.edu>
+ */
+
+#ifndef PULSE_COUNTER_H_
+#define PULSE_COUNTER_H_
+
+#include <stdint.h>
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @brief   Parameters needed for device initialization
+ */
+typedef struct {
+    gpio_t    gpio;            /**< GPIO pin that sensor is connected to */
+} pulse_counter_params_t;
+
+/**
+  * @brief   Device descriptor for a PULSE_COUNTER device
+  * @{
+  */
+typedef struct {
+    uint8_t   id;             /**< pulse counter id to distinguish multiple counters */
```

Comment:
I suggest putting the count itself here instead of an index to a static array.

Reply:
... or as a pointer as suggested by @cladmi

Reply:
Yeah, I will use a pointer! Thanks :)